src/Settings.php

```diff
@@ -0,0 +1,418 @@
+				],
+				'start_time'          => [
+					'type'            => 'dropdown',
+					'options'         => $start_hours,
+					'label'           => esc_html__( 'Start hour', 'tribe-ext-limit-week-view-time-range' ),
+					'tooltip'         => $this->get_start_time_tooltip(),
+					'default'         => '0',
+					'validation_type' => 'html',
+				],
+				'end_time'            => [
+					'type'            => 'dropdown',
+					'options'         => $end_hours,
+					'label'           => esc_html__( 'End hour', 'tribe-ext-limit-week-view-time-range' ),
+					'tooltip'         => $this->get_end_time_tooltip(),
+					'default'         => '24',
+					'validation_type' => 'html',
+				],
+				'sidebar_time_format' => [
+					'type'            => 'dropdown',
+					'options'         => $this->get_sidebar_time_format(),
+					'label'           => esc_html__( 'Time format', 'tribe-ext-limit-week-view-time-range' ),
+					'tooltip'         => $this->get_sidebar_time_format_tooltip(),
+					'default'         => 'g a',
+					'validation_type' => 'html',
+				],
+				'show_grid' => [
+					'type'            => 'checkbox_bool',
+					'label'           => esc_html__( 'Show grid', 'tribe-ext-limit-week-view-time-range' ),
+					'tooltip'         => $this->get_show_grid_tooltip(),
+					'validation_type' => 'boolean',
+				],
+			];
+
+			/**
+			 * Remove 'sidebar_time_format' if using V1
+			 */
+			if ( ! function_exists( 'tribe_events_views_v2_is_enabled' ) || ( function_exists( 'tribe_events_views_v2_is_enabled' ) && empty( tribe_events_views_v2_is_enabled() ) ) ) {
+				unset( $fields['sidebar_time_format'] );
+				unset( $fields['show_grid'] );
+			}
+
+			$this->settings_helper->add_fields( $this->prefix_settings_field_keys( $fields ),
+			                                    'display',
+			                                    'enable_month_view_cache',
+			                                    false );
+		}
+
+		private function get_sidebar_time_format() {
+			$time_format = [
+				'g a'   => '3 pm (default)',
+				'g:i a' => '3:00 pm',
+				'g A'   => '3 PM',
+				'G'     => '15',
+				'G:i'   => '15:00',
+				'Hi'    => '1500',
+			];
+
+			return $time_format;
+		}
+
+		/**
+		 * Add the options prefix to each of the array keys.
+		 *
+		 * @param array $fields
+		 *
+		 * @return array
+		 */
+		private function prefix_settings_field_keys( array $fields ) {
+			$prefixed_fields = array_combine( array_map( function ( $key ) {
+				return $this->get_options_prefix() . $key;
+			},
+				array_keys( $fields ) ),
+			                                  $fields );
+
+			return (array) $prefixed_fields;
+		}
+
+		/**
+		 * Here is an example of getting some HTML for the Settings Header.
+		 *
+		 * @return string
+		 */
+		private function get_setting_intro_text() {
+			$result = '<h3>' . esc_html_x( 'Limit Week View Time Range',
+			                               'Settings header',
+			                               'tribe-ext-limit-week-view-time-range' ) . '</h3>';
```

Comment:
here and elsewhere: `esc_html_x(` on its own line, following on the next line, and closing `)` on its own line.

Reply:
Done throughout all files.